DXMainClient/Domain/Multiplayer/MapLoader.cs

```diff
@@ -55,6 +88,132 @@ public void LoadMapsAsync()
+            // The way we're detecting the loaded map is hacky, but we don't
+            // have the SHA-1 to work with.
+            foreach (GameMode gameMode in GameModes)
+            {
+                gameMode.Maps.RemoveAll(map => Path.GetFileNameWithoutExtension(map.CompleteFilePath).EndsWith(name));
+            }
+
+            RemoveEmptyGameModesAndUpdateGameModeMaps();
+            GameModeMapsUpdated?.Invoke(null, new MapLoaderEventArgs(null));
+        }
+
+        /// <summary>
+        /// Handle a file being renamed in the custom map folder.
+        ///
+        /// If a file is renamed from "something.map" to "somethingelse.map" then there is a high likelyhood
+        /// that nothing will change in the client because the map data was already loaded.
+        ///
+        /// This is mainly here because Final Alert 2 will often export as ".yrm" which requires a rename.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        public void HandleCustomMapFolder_Renamed(object sender, RenamedEventArgs e)
+        {
+            string name = Path.GetFileNameWithoutExtension(e.Name);
+            string relativeMapPath = SafePath.CombineFilePath(CustomMapsDirectory, name);
+            bool oldPathIsMap = Path.GetExtension(e.OldName) == MAP_FILE_EXTENSION;
+            bool newPathIsMap = Path.GetExtension(e.Name) == MAP_FILE_EXTENSION;
+
+            // Check if the user is renaming a non ".map" file.
+            // This is just for logging to help debug.
+            if (!oldPathIsMap && newPathIsMap)
+            {
+                Logger.Log($"Renaming file changed the file extension. User is likely renaming a '.yrm' from Final Alert 2: old={e.OldName}, new={e.Name}");
+            }
+            else if (oldPathIsMap && !newPathIsMap)
+            {
+                // A bit hacky, but this is a rare case.
+                Logger.Log($"Renaming file changed the file extension to no longer be '.map' for some reason, removing from map list: old={e.OldName}, new={e.Name}");
+                HandleCustomMapFolder_Deleted(sender, e);
+            }
+
+            if (!newPathIsMap)
+            {
+                Logger.Log($"Renaming file. New extension is not '{MAP_FILE_EXTENSION}', moving on: file={e.Name}");
+                return;
+            }
+
+            Map map = LoadCustomMap(relativeMapPath, out string result);
+
+            if (map == null)
+            {
+                Logger.Log($"Failed to load renamed map file. Map is likely already loaded: original={e.OldName}, new={e.Name}, reason={result}");
+            }
+        }
+
+        /// <summary>
+        /// Handle errors in the filewatcher.
+        ///
+        /// Not much to do other than log a stack trace.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        public void HandleCustomMapFolder_Error(object sender, ErrorEventArgs e)
```

Comment:
As far as I can see, this is something that should never be called outside of this class. Make it `private`. The same also applies to other file watcher event handlers.

Reply:
Made all private